superset-frontend/src/components/Menu/MenuObject.tsx

```diff
@@ -51,22 +61,20 @@ export default function MenuObject({ label, icon, childs, url, index }) {
       eventKey={index}
       title={navTitle}
     >
-      {childs.map((child, index1) =>
-        child === '-' ? (
+      {childs?.map((child: MenuObjectChildProps | string, index1: number) =>
+        typeof child === 'string' && child === '-' ? (
           <MenuItem key={`$${index1}`} divider />
         ) : (
           <MenuItem
-            key={`${child.label}`}
-            href={child.url}
+            key={`${(child as MenuObjectChildProps).label}`}
+            href={(child as MenuObjectChildProps).url}
             eventKey={parseFloat(`${index}.${index1}`)}
           >
-            <i className={`fa ${child.icon}`} />
-            &nbsp; {child.label}
+            <i className={`fa ${(child as MenuObjectChildProps).icon}`} />
+            &nbsp; {(child as MenuObjectChildProps).label}
```

Comment:
Maybe I was a bit unclear. It'd be nice to get rid of all the casting here entirely. Maybe something this will remove the need for all casting:
```tsx
{childs?.map((child: MenuObjectChildProps | string, index1: number) => {
  if (typeof child === 'string' && child === '-') {
    return <MenuItem key={`$${index1}`} divider />;
  } else if (typeof child !== 'string') {
    return (
      <MenuItem
        key={child.label}
        href={child.url}
        ...
      </MenuItem>
    );
  else {
    return null;
  }
}
```

Reply:
@etr2460 , I didn't do it because of the pending conversation of changing the type of childs. Once that is resolved, I agree, that casting will not be required at all

Reply:
Gotcha. I think the typing is correct above as is (and if it's not, then we can always improve it in the future)

Reply:
Done. removed the type casting and replaced it with type check

Reply:
The above condition gives linting errors:
  65:9  error  Expected an assignment or function call and instead saw an expression  no-unused-expressions
  65:9  error  Do not nest ternary expressions                                        no-nested-ternary